bbb/services.py

```diff
@@ -266,31 +267,44 @@ def start(self):
             self.reflectTasks()
             time.sleep(self.interval)
 
+    def _handle_taskcluster_exceptions(self, t, exc):
+        status_code = exc.superExc.response.status_code
+
+        if status_code == 409:
+            log.info("Deadline exceeded for task %s run %s, cancelling it",
+                     t.taskId, t.runId)
+            branch = self.buildbot_db.getBranch(t.buildrequestId).split("/")[-1]
+            try:
+                for id_ in self.buildbot_db.getBuildIds(t.buildrequestId):
+                        self.selfserve.cancelBuild(branch, id_)
+            except RequestException:
+                log.exception("Failed to cancel task %s run %s build request ID %s",
+                              t.taskId, t.runId, t.buildrequestId)
+            else:
+                # delete from the DB only if all cancel requests pass
+                self.bbb_db.deleteBuildRequest(t.buildrequestId)
```

Comment:
I think this is correct...but would this make sense as the last line inside the try: section above?